django/db/models/base.py

```diff
@@ -883,13 +882,12 @@ def _do_update(self, base_qs, using, pk_val, values, update_fields, forced_updat
             )
         return filtered._update(values) > 0
 
-    def _do_insert(self, manager, using, fields, update_pk, raw):
+    def _do_insert(self, manager, using, fields, raw):
         """
         Do an INSERT. If update_pk is defined then this method should return
         the new pk for the model.
         """
-        return manager._insert([self], fields=fields, return_id=update_pk,
-                               using=using, raw=raw)
+        return manager._insert([self], fields=fields, using=using, raw=raw)
```

Comment:
Is there a reason not to favor the previous approach and pass `returning_fields=self._meta.returning_fields` instead? That seems like a better separation of concerns to me than having the insert compiler lookup `returning_fields`.

Reply:
What do you mean by previous approach? The argument `rerturn_id` does still exist, but for this call it remains the default.

I honestly don't want to change the API here, I want to keep the changeset as small as possible.

Reply:
I tried to address Simon's request in the 13477049aa6b959a98d6fb941a08edfa13046b31, this change also replaces some misleading variables' names.

Reply:
@charettes Would you like to take a look?

Reply:
Just one comment. If we go down that path, does that mean, we should also pass the table name in the future? The options are already available to the compiler for that very reason. I don't say it should be done within this patch – in fact I would be against it – but it would make this more consistent and decouple both layers.

Reply:
I don't think that we need to pass a table name because `_insert()` is a `QuerySet` method so it already contains `self.model` with a table name :thinking: but yeah let's leave it for a separate discussion.